test/Bitcoin/UtilSpec.hs

```diff
@@ -88,3 +107,125 @@ readTestFile fp =
     A.eitherDecodeFileStrict ("data/" <> fp) >>= either (error . message) return
   where
     message aesonErr = "Could not read test file " <> fp <> ": " <> aesonErr
+
+
+-- Helpers for creating Serial and JSON Identity tests
+
+data SerialBox
+    = forall a.
+        (Show a, Eq a, Typeable a, Serial a) =>
+      SerialBox (Gen a)
```

Comment:
These existential constructors are a bit of a smell and adds needless cognitive overhead.  We can put the constraints on the testing function and just pass in the `Gen a` values directly.

Reply:
Upon reviewing this further. The changes to support this are going to be quite substantial and should be done as we revamp the test suite.